ted/src/main/java/frc/robot/subsystems/LEDSubsystem.java

```diff
@@ -0,0 +1,166 @@
+        if(currentActions.containsKey(LEDState.Green) && currentActions.get(LEDState.Green).booleanValue()) {
+              targetLedState = LEDState.Green;
+        }
+        else if(currentActions.containsKey(LEDState.Yellow) && currentActions.get(LEDState.Yellow).booleanValue()) {
+              targetLedState = LEDState.Yellow;
+        }
+        else if(currentActions.containsKey(LEDState.OrangeSolid) && currentActions.get(LEDState.OrangeSolid).booleanValue()) {
+              targetLedState = LEDState.OrangeSolid;
+        }
+        else if(currentActions.containsKey(LEDState.OrangeBlink) && currentActions.get(LEDState.OrangeBlink).booleanValue()) {
+              targetLedState = LEDState.OrangeBlink;
+        }
+        // update the LED state when the target state has changed
+        if(this.currentLEDState != targetLedState) {
+              this.currentLEDState = targetLedState;
+              if(this.currentLEDState == LEDState.Green) {
+                    this.greenSolid();
+              }
+              else if(this.currentLEDState == LEDState.Yellow) {
+                    this.yellowSolid();
+              }
+              else if(this.currentLEDState == LEDState.OrangeSolid) {
+                    this.orangeSolid();
+              }
+              else if(this.currentLEDState == LEDState.OrangeBlink) {
+                    this.orangeBlink();
+              }
+              else if(this.currentLEDState == LEDState.Off) {
+                    this.offState();
+              }
+              System.out.println("**** UPDATING LED STATE TO " + this.currentLEDState.toString());
+        }
+        else if(this.lastBlinkState != this.currentBlinkState && this.currentLEDState == LEDState.OrangeBlink) {
+              System.out.println("**** BLINKING LED STATE TO " + this.currentLEDState.toString());
+              this.orangeBlink();
+        }
+  }
+
+      //Sets leds to orange blink
+      private void orangeBlink() {
+            if(this.currentBlinkState) {
+                  this.setLedStringColor(255,165,0);
+            }
+            else {
+                  this.setLedStringColor(0, 0, 0);
+            }
+      }
+
+      //Sets leds to orange solid
+      private void orangeSolid() {
+            this.setLedStringColor(255,165,0);
+      }
+
+      //Sets leds to yellow solid
+      private void yellowSolid() {
+            this.setLedStringColor(150,150,0);
+      }
+
+      //Sets leds to green solid
+      private void greenSolid() {
+            this.setLedStringColor(0,200,0);
+      }
+
+      //Turns off leds
+      private void offState() {
+            this.setLedStringColor(0,0,0);
+            this.leds.modulateVBatOutput(0);
+      }
+
+      //Sets leds color
+      private void setLedStringColor(int red, int green, int blue) {
+                  this.leds.setLEDs(red, green, blue, 0, startIdx, BUFFER_LENGTH);
+                  this.leds.modulateVBatOutput(0.9);
```

Comment:
above, you set the brightness to 0.5 and here you are modulating the output voltage by 0.9. How do these effects interact? Maybe choose one way of modulating the brightness and then make that a constant in Constants.java

Reply:
done